schematic/models/validate_attribute.py

```diff
@@ -868,186 +883,590 @@ def url_validation(
+            validation_rows.append(entry.index[0] + 2)
+            validation_values.append(entry.values[0])
+
+        # Parse invalid_rows and invalid_values so they can be used to raise warnings/errors
+        invalid_rows = iterable_to_str_list(set(validation_rows))
+        invalid_entries = iterable_to_str_list(set(validation_values))
+
+        return invalid_rows, invalid_entries, manifest_ids
+
+    def _merge_format_invalid_rows_values(
+        series_1: pd.core.series.Series, series_2: pd.core.series.Series
+    ) -> tuple([[], []]):
+        """Merge two series to identify gather all invalid values, and parse out invalid rows and entries
+        Args:
+            series_1, pd.core.series.Series: first set of invalid values to extract
+            series_2, pd.core.series.Series: second set of invalid values to extract
+        Returns:
+            invalid_rows, list: invalid rows taken from both series
+            invalid_entry, list: invalid values taken from both series
+        """
+        # Merge series to get invalid values and rows
+        invalid_values = pd.merge(series_1, series_2, how="outer")
+        # Increment pandas row values by 2 so they will match manifest sheet row values.
+        invalid_rows = (
+            pd.merge(
+                series_1,
+                series_2,
+                how="outer",
+                left_index=True,
+                right_index=True,
+            ).index.to_numpy()
+            + 2
+        )
+        # Parse invalid_rows and invalid_values so they can be used to raise warnings/errors
+        invalid_rows = np_array_to_str_list(invalid_rows)
+        invalid_entry = iterable_to_str_list(invalid_values.squeeze())
+        return invalid_rows, invalid_entry
+
+    def _format_invalid_row_values(
+        invalid_values: Dict[str, pd.core.series.Series]
+    ) -> tuple([[], []]):
+        """Parse invalid_values dictionary, to extract invalid_rows and invalid_entry to be used later
+        to raise warnings or errors.
+        Args:
+            invalid_values, Dict[str, pd.core.series.Series]:
+        Returns:
+            invalid_rows, list: invalid rows recorded in invalid_values
+            invalid_entry, list: invalid values recorded in invalid_values
+        """
+        # Increment pandas row values by 2 so they will match manifest sheet row values.
+        invalid_rows = invalid_values.index.to_numpy() + 2
+
+        # Parse invalid_rows and invalid_values so they can be used to raise warnings/errors
+        invalid_rows = np_array_to_str_list(invalid_rows)
+        invalid_entry = iterable_to_str_list(invalid_values)
+        return invalid_rows, invalid_entry
+
+    def _gather_set_warnings_errors(
         val_rule: str,
-        manifest_col: pd.core.series.Series,
-        project_scope: list,
+        source_attribute: str,
+        set_validation_store: list[{}, [], {}],
```

Comment:
`[{}, [], {}]` isn't a valid type. If its always three things as described belowe perhaps it should be a tuple instead of a list.